src/System.Slices/src/System/Contract.cs

```diff
@@ -31,17 +31,26 @@ public static void Requires(bool condition)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static void RequiresInRange(int start, int length)
         {
-            if (!(start >= 0 && start < length))
+            // explicit unchecked block is used in order to avoid OverflowException when casting 
+            // negative int to unsigned int when the code is compiled with /checked option
+            unchecked
             {
-                throw new ArgumentOutOfRangeException();
+                // use unsigned int to reduce range check by one ( >= 0 )
+                if ((uint)start >= (uint)length)
```

Comment:
BTW, from practical perspective, I would add the check for >0 and do a commit. Then create a separate PR to remove the check if we see the benefit/risk being large.